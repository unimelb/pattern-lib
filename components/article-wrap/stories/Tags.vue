<template>
  <div class="tags">
    <div class="tags__heading">
      <SvgIcon
        aria-label="Tags"
        class="push-icon__icon"
        width="15px"
        height="15px"
        name="tag" />
    </div>
    <div class="tags__listing">
      <a href="">tag</a>,
      nonlinked,
      <a
        class="tag"
        href="">multi word</a>,
      <a
        class="tag tag--truncate"
        href=""
        title="tag truncated because it's way too long">tag truncated because it's way too long</a>
    </div>
  </div>
</template>

<script>
import SvgIcon from '../../icons/SvgIcon.vue';

export default {
  components: {
    SvgIcon,
  },
};
</script>
