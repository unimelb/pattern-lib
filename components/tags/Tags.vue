<template>
  <div class="tags">
    <ul class="tags__list">
      <li
        v-for="(item, index) in data"
        :key="index"
        class="tags__item list-reset">
        <a
          v-if="item.href"
          :href="item.href"
          class="tags__link">
          {{ item.value }}
        </a>
        <span v-else>{{ item.value || item }}</span>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    data: {
      type: Array,
      default: () => [],
    },
  },
};
</script>
