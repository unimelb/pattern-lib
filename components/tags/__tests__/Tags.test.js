import { shallow } from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import Tags from '../Tags.vue';

expect.extend(toHaveNoViolations);


describe('Tags', () => {
  it('should match snapshot', () => {
    const result = shallow(Tags).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(Tags).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
