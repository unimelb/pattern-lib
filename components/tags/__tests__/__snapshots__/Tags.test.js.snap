// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should match snapshot 1`] = `
<div
  class="tags"
>
  <ul
    class="tags__list"
  />
</div>
`;
